addons/sale/wizard/sale_make_invoice_advance.py

```diff
@@ -354,22 +354,31 @@ def _prepare_down_payment_lines_values(self, order):
                 ])
 
         downpayment_line_map = {}
+        analytic_map = {}
         for tax_id, analytic_distribution, price_subtotal in down_payment_values:
-            grouping_key = frozendict({
-                'tax_id': tuple(sorted(tax_id.ids)),
-                'analytic_distribution': analytic_distribution,
-            })
+            grouping_key = frozendict({'tax_id': tuple(sorted(tax_id.ids))})
             downpayment_line_map.setdefault(grouping_key, {
                 **base_downpayment_lines_values,
                 **grouping_key,
                 'product_uom_qty': 0.0,
                 'price_unit': 0.0,
             })
             downpayment_line_map[grouping_key]['price_unit'] += price_subtotal
-        for key in downpayment_line_map:
-            downpayment_line_map[key]['price_unit'] = \
-                order.currency_id.round(downpayment_line_map[key]['price_unit'] * percentage)
-
+            if analytic_distribution:
+                analytic_map.setdefault(grouping_key, [])
+                analytic_map[grouping_key].append((price_subtotal, analytic_distribution))
+
+        for key, line_vals in downpayment_line_map.items():
+            # weight analytic account distribution
+            if analytic_map.get(key):
+                analytic_distribution = {}
+                for sub_price, ana_account_distrib in analytic_map[key]:
```

Comment:
how about naming that would be a bit more consistent
```suggestion
                line_analytic_distribution = {}
                for price_subtotal, analytic_distribution in analytic_map[key]:
                    for account, distribution in analytic_distribution.items():
```